src/Rotations.jl

```diff
@@ -0,0 +1,148 @@
+@doc doc"""
+    Rotations{N} <: Manifold
+
+Special orthogonal manifold $\mathrm{SO}(n)$ represented by $n\times n$
+real-valued orthogonal matrices with determinant $+1$.
+
+# Constructor
+
+    Rotations(n, retract! = rectract_qr!)
+
+generates the $\mathrm{SO}(n) \subset \mathbb R^{n\times n}$
+"""
+struct Rotations{N, TRetr<:Function} <: Manifold
+    retract!::TRetr
+end
+Rotations(n::Int, retract!::TRetr = retract_qr!) where TRetr = Rotations{n,TRetr}(retract!)
+
+@doc doc"""
+    dimension(S::Rotations)
+
+returns the dimension of the manifold $\mathrm{SO}(n)$, i.e. $\frac{n(n-1), 2}$.
+"""
+dimension(S::Rotations{N}) where {N} = div(N*(N-1), 2)
+
+@doc doc"""
+    inner(S,x,w,v)
+
+compute the inner product of the two tangent vectors `w,v` from the tangent
+plane at `x` on the special orthogonal space `S=`$\mathrm{SO}(n)$ using the
+restriction of the metric from the embedding, i.e. $ (v,w)_x = \frac{1,2}v^\mathrm{T}w $.
+"""
+inner(S::Rotations, x, w, v) = dot(w, v)/2
```

Comment:
Would be good to have documentation on what `v` and `w` represent. I understand in the context of Lie algebras but not in the context of an embedded manifold. How is the factor of 1/2 a consequence of the embedding?

Reply:
I've improved the documentation, is it OK now?

Reply:
It's more clear regarding tangent vectors now. It still seems that the  factor of 1/2 is chosen pragmatically, not as a consequence of the induced metric of the embedding, which is confusing. Or perhaps this is not an isometric embedding, where the Euclidean space is scaled by 1/sqrt(2) before the rotation matrices are embedded in it. I hadn't seen a discussion of this in Absil's book before, but I'll check again.

Reply:
I've taken the factor 1/2 from this book actually: https://www.springer.com/gp/book/9781493940189 . Maybe it's a bad choice after all.